include/odp/api/spec/packet_io.h

```diff
@@ -121,6 +121,10 @@ typedef union odp_pktin_hash_proto_t {
 		uint32_t ipv6_tcp : 1;
 		/** IPv6 addresses */
 		uint32_t ipv6     : 1;
+		/** Destination MAC address */
+		uint32_t dmac	  : 1;
```

Comment:
If the HW is capable of including destination MAC address and upper layer fields (like IP addresses, L4 ports) in the same hash, then that is what I might want to use to get even more flow entropy. But how could I express that? Having only the destination MAC address in the flow hash seems a bit limiting to me.

The current hash config looks a bit weird to me since, despite being a bit field, it tries to specify the hash strategies separately (e.g. ipv4_tcp) instead of as desired fields to include in the hash. So in keeping with that the dmac combined with other fields would give rise to quite many new combinations. Maybe one possibility would be to move dmac to a separate "layer" of the config so that the new "L2 layer hash config" and the current "L3+L4 layer hash config" would be combined.

I am not sure dmac is a good name. dst_mac would sound more clear to me.

Maybe there should be source MAC too.